Diffusion-macOS/ControlsView.swift

```diff
@@ -47,345 +65,405 @@ struct LabelToggleDisclosureGroupStyle: DisclosureGroupStyle {
-                    } label: {
-                        HStack {
-                            Label("Prompts", systemImage: "text.quote").foregroundColor(.secondary)
-                            Spacer()
-                            if disclosedPrompt {
-                                Button {
-                                    showPromptsHelp.toggle()
-                                } label: {
-                                    Image(systemName: "info.circle")
-                                }
-                                .buttonStyle(.plain)
-                                // Or maybe use .sheet instead
-                                .popover(isPresented: $showPromptsHelp, arrowEdge: .trailing) {
-                                    promptsHelp($showPromptsHelp)
-                                }
-                            }
-                        }.foregroundColor(.secondary)
-                    }
-                    Divider()
-
-                    let guidanceScaleValue = generation.guidanceScale.formatted("%.1f")
-                    DisclosureGroup(isExpanded: $disclosedGuidance) {
-                        CompactSlider(value: $generation.guidanceScale, in: 0...20, step: 0.5) {
-                            Text("Guidance Scale")
-                            Spacer()
-                            Text(guidanceScaleValue)
-                        }.padding(.leading, 10)
-                    } label: {
-                        HStack {
-                            Label("Guidance Scale", systemImage: "scalemass").foregroundColor(.secondary)
-                            Spacer()
-                            if disclosedGuidance {
-                                Button {
-                                    showGuidanceHelp.toggle()
-                                } label: {
-                                    Image(systemName: "info.circle")
-                                }
-                                .buttonStyle(.plain)
-                                // Or maybe use .sheet instead
-                                .popover(isPresented: $showGuidanceHelp, arrowEdge: .trailing) {
-                                    guidanceHelp($showGuidanceHelp)
+            let filledCircle = Image(systemName: "circle.fill")
+                .font(.caption)
+                .foregroundColor(exists ? .accentColor : .secondary)
+            
+            let dottedCircle = Image(systemName: "circle.dotted")
+                .font(.caption)
+                .foregroundColor(exists ? .accentColor : .secondary)
+            
+            let dl = Image(systemName: "arrow.down.circle")
+                .font(.caption)
+                .foregroundColor(.gray)
+            
+            
+            return HStack {
+                if model.builtin && !exists {
+                    dl
+                } else if exists {
+                    filledCircle
+                } else {
+                    dottedCircle
```

Comment:
what does this state represent? The model is not builtin and does not exist - is it an invalid folder?

Reply:
In the second check we're not checking against builtin. If the model exists. Meaning if it's on the local filesystem. It could be either built-in or 3rd party all we care here is - you can continue because it's locally available.

In the third check we know it's not a built-in model that's missing and it's not present on the system. This is an unknown / broken state for sure. Rather than ignore I wanted to indicate "something's amiss" which would allow the user to manually intervene at least.

Reply:
Ok. What do you think about using some other icon that looks more like an error, like a red `×` or something?

Reply:
Agreed Also any edge cases where a built-in model is downloaded and corrupt should be identified and handled instead of simply shown as an error. They're some missing checks of state I think. I do think file system awareness should be moved into functions in the ModelInfo itself or into a Model ViewModel management object to centralise the logic on state checks, etc.